pkg/controller/gitopsconfig/controller.go

```diff
@@ -538,70 +543,56 @@ func watchJobForStatus(r *Reconciler, jobName string, namespace string, instance
 }
 
 func scheduleStatusForCronJobs(job jobdata, schedule string, instance *gitopsv1alpha1.GitOpsConfig) {
-	var result batchv1beta1.CronJob
-	err := job.Reconcile.client.Get(context.TODO(),
-		types.NamespacedName{Name: job.Name, Namespace: job.Namespace},
-		&result)
+	result, err := job.getCronJobData()
 
 	if err != nil {
-		log.Info("Error in GetCronJob:" + err.Error())
+		log.Error(err, "Error in GetCronJob")
 		return
 	}
 
 	instance.Status = gitopsv1alpha1.GitOpsConfigStatus{}
-	if len(result.Status.Active) < 1 {
+	if len(result.Status.Active) == 0 {
 		log.Info("There is no active job")
-		if result.Status.LastScheduleTime != nil {
-			instance.Status.LastScheduleTime = result.Status.LastScheduleTime.Format(time.RFC3339)
-		}
-		instance.Status.Message = "There is no active job"
-		err = job.Reconcile.client.Status().Update(context.TODO(), instance)
+
+		err = job.updateStatus(instance, gitopsv1alpha1.GitOpsConfigStatus{
+			LastScheduleTime: result.Status.LastScheduleTime,
+			Message:          "There is no active job",
```

Comment:
What would you think about adding a line here like below?

    // TODO: set State and other fields based on most recent Job

Reply:
i dont think we need to add mentioned line.. because if `active = 0` it has only Last schedule time in job status